### DISCUSSION TITLE
Aliases in catalog/TIMESERIES

### DISCUSSION THREAD

**Discussion by mdstanfill** [USACE/cwms-data-api#795] [General]
I see that `catalog/TIMESERIES` _can_ return a list of aliases for a given time series, but I'm having a couple of issues with it.

Primarily, it doesn't seem to return a timeseries if you ask for it by an alias, e.g.:
```
curl -X 'GET' \
  'https://wm.nwp.ds.usace.army.mil:8243/nwdp-data-314rc4/catalog/TIMESERIES?office=NWDP&like=DET_S1-D20ft.Temp-Water.%2ABest' \
  -H 'accept: application/json;version=2'
```
doesn't find `DET_S1-D20ft.Temp-Water.Inst.0.0.Best`, an alias for `DET_S1-D20ft.Temp-Water.Inst.0.0.MIXED-REV`, instead returning:
```
{
  "entries": [],
  "page-size": 500,
  "total": 0
}
```
Additionally, (of no immediate concern, but for informational purposes), even if I ask for `MIXED-REV`, it doesn't show `Best` as an alias.  E.g.:
```
curl -X 'GET' \
  'https://wm.nwp.ds.usace.army.mil:8243/nwdp-data-314rc4/catalog/TIMESERIES?office=NWDP&like=DET_S1-D20ft.Temp-Water.%2AMIXED-REV' \
  -H 'accept: application/json;version=2'
```
returns
```
{
  "entries": [
    {
      "office": "NWDP",
      "name": "DET_S1-D20ft.Temp-Water.Inst.0.0.MIXED-REV",
      "units": "C",
      "interval": "0",
      "interval-offset": -2147483648,
      "time-zone": "US/Pacific",
      "extents": [
        {
          "earliest-time": "2010-06-02T20:00:00Z",
          "last-update": "2024-07-17T22:42:48.548Z",
          "latest-time": "2024-07-17T22:00:00Z"
        }
      ]
    }
  ],
  "page-size": 500,
  "total": 1
}
```
with no `aliases` attribute.

**Comment by rma-rripken** [comment]
The Location Catalog does do something with the aliases but the Timeseries Catalog does not.  The entries in the the Timeseries Catalog results do not have a field to hold aliases and the queries do not attempt to specifically include or exclude them.
